src/Features/Core/Portable/EmbeddedLanguages/Classification/AbstractFallbackEmbeddedLanguageClassifier.cs

```diff
@@ -45,8 +50,15 @@ public void RegisterClassifications(EmbeddedLanguageClassificationContext contex
 
             foreach (var vc in virtualChars)
             {
+                // non-BMP UTF-16 characters' length can be two
                 if (vc.Span.Length > 1)
-                    context.AddClassification(ClassificationTypeNames.StringEscapeCharacter, vc.Span);
+                {
+                    var text = token.SyntaxTree?.GetText(context.CancellationToken).ToString(vc.Span);
```

Comment:
this allocates on every character, which is not something we want.  Why can't we just examine the characters directly at that text location?